### PR TITLE
Skeleton of CSP integration in mod_pagespeed.

### PR DESCRIPTION
**Pull request by morlovich**
Hooks up CSP with RewriteDriver and ScanFilter and an off switch. Makes one filter use it.

### CODE REVIEW

File: net/instaweb/rewriter/responsive_image_filter.cc [hunk 1]
```diff
@@ -432,9 +432,17 @@ void ResponsiveImageSecondFilter::Cleanup(
 
 void ResponsiveImageSecondFilter::EndDocument() {
   if (zoom_filter_enabled_ && srcsets_added_ && !driver()->is_amp_document()) {
-    HtmlElement* script = driver()->NewElement(NULL, HtmlName::kScript);
-    driver()->AddAttribute(script, HtmlName::kSrc, responsive_js_url_);
-    InsertNodeAtBodyEnd(script);
+    if (driver()->IsLoadPermittedByCsp(
+          CspDirective::kScriptSrc, responsive_js_url_)) {
+      HtmlElement* script = driver()->NewElement(NULL, HtmlName::kScript);
```
Comment (jmarantz) [inline net/instaweb/rewriter/responsive_image_filter.cc]: nullptr

**Comment by oschaaf** [comment]
great, lgtm!

**Review by jmarantz — APPROVED** [review]
at some point will there be a system test?

File: net/instaweb/rewriter/responsive_image_filter.cc (hunk 1 above)
Comment (morlovich) [reply net/instaweb/rewriter/responsive_image_filter.cc]: Done; doesn't seem like I have an integration test in any of the outstanding CLs, though.

**Pull request merged**